simphony_ui/ui.py

```diff
@@ -0,0 +1,199 @@
+    young_modulus = Array(Float, (2,), [2.e4, 2.e4])
+    poisson_ratio = Array(Float, (2,), [0.45, 0.45])
+    restitution_coefficient = Array(Float, (4,), [0.95, 0.95, 0.95, 0.95])
+    friction_coefficient = Array(Float, (4,), [0.0, 0.0, 0.0, 0.0])
+    cohesion_energy_density = Array(Float, (4,), [0.0, 0.0, 0.0, 0.0])
+
+    # TODO: Check in the documentation if there must be more
+    # possibilities in dropdowns
+    pair_potentials_1 = Enum('repulsion', 'cohesion')
+    pair_potentials_1.default_value = 'repulsion'
+    pair_potentials_2 = Enum('repulsion', 'cohesion')
+    pair_potentials_2.default_value = 'cohesion'
+
+    # TODO: label is not displayed properly: fix it
+    pair_potentials = HGroup(
+        Item(name='pair_potentials_1', show_label=False),
+        Item(name='pair_potentials_2', show_label=False),
+        label='Pair potentials'
+    )
+
+    traits_view = View(
+        VGroup(
+            Item(name='timestep'),
+            Item(name='num_iterations', label='Number of iterations'),
+            '_',
+            Item(name='input_file'),
+            '_',
+            boundary_condition,
+            '_',
+            Item(name='young_modulus'),
+            Item(name='poisson_ratio'),
+            Item(name='restitution_coefficient'),
+            Item(name='friction_coefficient'),
+            Item(name='cohesion_energy_density'),
+            pair_potentials,
+        )
+    )
+
+
+class OpenfoamModel(HasStrictTraits):
+    """ The model of Openfoam input parameters """
+
+    mode = Enum('internal', 'io')
+    mode.default_value = 'internal'
+
+    # TODO: input_file = ???
+    mesh_name = Str('mesh')
+    mesh_type = Enum('block', 'quad')
+    mesh_type.default_value = 'block'
+
+    output_path = Directory(os.path.abspath(os.path.curdir))
+
+    timestep = PositiveFloat(2.0e-4)
+    num_iterations = PositiveInt(10)
+
+    viscosity = PositiveFloat(1.0e-3)
+    density = PositiveFloat(1000.0)
+    delta_p = Float(0.008)
+
+    # TODO: boundary_conditions_velocity = ???
+    # TODO: boundary_conditions_pressure = ???
+
+    # TODO: PosInt and PosFloat not working for Arrays....
+    # TODO: Try to display arrays on only one line
+    channel_size = Array(PositiveFloat, (3,), [1.0e-1, 1.0e-2, 2.0e-3])
+    num_grid = Array(PositiveInt, (3,), [400, 40, 1])
+
+    traits_view = View(
+        VGroup(
+            Item(name='mode', style='custom'),
+            '_',
+            Item(name='mesh_name'),
+            Item(name='mesh_type'),
+            '_',
+            Item(name='output_path'),
+            '_',
+            Item(name='timestep'),
+            Item(name='num_iterations', label='Number of iterations'),
+            '_',
+            Item(name='viscosity'),
+            Item(name='density'),
+            Item(name='delta_p'),
+            '_',
+            Item(name='channel_size'),
+            Item(name='num_grid'),
+        )
+    )
+
+
+class TabsModel(HasStrictTraits):
```

Comment:
Well, this is not really a "TabsModel". it's your application. I would call it MainWindow, or Application.